src/main/java/com/hubspot/jinjava/lib/fn/eager/EagerMacroFunction.java

```diff
@@ -142,13 +142,24 @@ public String reconstructImage() {
       return "";
     } else {
       try {
+        int numEagerTokensStart = interpreter.getContext().getEagerTokens().size();
         String evaluation = (String) evaluate(
           macroFunction
             .getArguments()
             .stream()
             .map(arg -> DeferredMacroValueImpl.instance())
             .toArray()
         );
+        if (interpreter.getContext().getEagerTokens().size() > numEagerTokensStart) {
+          evaluation =
```

Comment:
Run the evaluation again now that all the variables that we attempted to modify have been deferred. This makes it so that recursion will work properly and those modified variables will be properly deferred within the reconstructed image.